src/Evolution/DgSubcell/PerssonTci.cpp

```diff
@@ -76,7 +74,11 @@ bool persson_tci_impl(const gsl::not_null<DataVector*> filtered_component,
     // to pow<double,double> already provides a certain amount of speed-up, it
     // would be sufficient to stick to the case 2 for now.
     //
-    if (pow(dg_mesh.extents(d) - 1, alpha) * l2Norm(*filtered_component) >
+
+    const double component_norm{l2Norm(component)};
+
+    if (pow(dg_mesh.extents(d) - num_highest_modes, alpha) *
```

Comment:
You should have a check somewhere (probably in option parsing/initialization) that `num_highest_modes < dg_mesh.extents(i)` for all dims. Otherwise this will likely FPE